geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/utils/GeoMessageSerializer.scala

```diff
@@ -89,6 +94,13 @@ object GeoMessageSerializer {
     val (serializer, version) = serialization match {
       case SerializationType.KRYO => (kryoSerializer, KryoVersion)
       case SerializationType.AVRO => (avroSerializer, AvroVersion)
+      case SerializationType.CONFLUENT =>
+        val confluentBuilder = schemaRegistryUrl match {
+          case Some(srUrl) => ConfluentFeatureSerializer.builder(sft, srUrl).withoutId.withUserData.immutable
+          case None =>
+            throw new IllegalArgumentException("Confluent serialization used, however no schema-registry url provided.")
+        }
+        (confluentBuilder.build(), ConfluentVersion)
```

Comment:
How about we just make it -1 to indicate that it is not used? The deserializer matches on the header number to determine the serialization type, so we would want it to be unique (i.e. not 1, 2 or 3). Since confluent is read-only for now, it doesn't really matter, but would help avoid confusion.

Reply:
Sounds good, thanks.